go/vt/vttablet/tabletserver/vstreamer/main_test.go

```diff
@@ -0,0 +1,368 @@
+/*
+Copyright 2019 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vstreamer
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"vitess.io/vitess/go/mysql/replication"
+	"vitess.io/vitess/go/vt/log"
+	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
+	"vitess.io/vitess/go/vt/vttablet/tabletserver/throttle/throttlerapp"
+
+	"github.com/stretchr/testify/require"
+
+	_flag "vitess.io/vitess/go/internal/flag"
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/vt/dbconfigs"
+	"vitess.io/vitess/go/vt/vtenv"
+	"vitess.io/vitess/go/vt/vttablet/tabletserver/tabletenv"
+	"vitess.io/vitess/go/vt/vttablet/tabletserver/vstreamer/testenv"
+)
+
+var (
+	engine *Engine
+	env    *testenv.Env
+
+	ignoreKeyspaceShardInFieldAndRowEvents bool
+	testRowEventFlags                      bool
+)
+
+func TestMain(m *testing.M) {
+	_flag.ParseFlagsForTest()
+	ignoreKeyspaceShardInFieldAndRowEvents = true
```

Comment:
Should we reset this to what it was before in a defer, in case it's used across tests?

Reply:
This is `TestMain`, not an individual test. This is the default setting overridden only in a few tests whre we test those attributes.